### PR TITLE
Fix callback handlers has function

### PR DESCRIPTION
**Pull request by pandamicro**
Re: cocos-creator/fireball#2577

Changes proposed in this pull request:
- 解决 Callback handlers 的 has 在 remove 和 invoke 穿插调用时的问题

> Makes sure these boxes are checked before submitting your PR - thank you!
> - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
> - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
> - [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
> - [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.

@cocos-creator/engine-admins

### CODE REVIEW

**Comment by pandamicro** [comment]
思来想去，还是觉得这部分的性能要保障，所以尽量避免了一切对象创建，用简单数组来管理。这样也更利于检索

File: cocos2d/core/platform/callbacks-invoker.js [hunk 1]
```diff
@@ -36,6 +36,35 @@ var CallbacksHandler = (function () {
     this._callbackTable = {};
 });
 
+// Avoid to equal to user set target (null for example)
+var REMOVE_PLACEHOLDER = Number.POSITIVE_INFINITY;
```
Comment (jareguo) [inline cocos2d/core/platform/callbacks-invoker.js:40]: 这里建议定义成 `var REMOVE_PLACEHOLDER = {};`

Comment (pandamicro) [inline cocos2d/core/platform/callbacks-invoker.js:40]: 为啥？

**Comment by pandamicro** [comment]
https://github.com/cocos-creator/engine/pull/714/commits/222458f453290d9f16b27d4e5770dec7345bba10 is for fixing bug in https://github.com/cocos-creator/engine/pull/699

File: cocos2d/core/platform/callbacks-invoker.js (hunk 1 above)
Comment (jareguo) [inline cocos2d/core/platform/callbacks-invoker.js:40]: 因为这样代码比较简短，效果一样，而且类型同样是 object

**Pull request merged**